### PR TITLE
gh-113971: Make `zipfile.ZipInfo._compresslevel` public as `.compress_level`

### LINKED ISSUE #113971: `zipfile.ZipInfo._compresslevel` should be public.
# Feature or enhancement

### Proposal:

Today people pass `ZipInfo` instances into APIs that accept them such as `ZipFile.writestr()` in order to control how individual items are put into a zip archive as such:

```python
    zip_info = zipfile.ZipInfo(filename=filename, date_time=desired_timestamp)
    zip_info.compress_type = this_files_compress_type
    # This attribute is sadly not public; manual creation of ZipInfo objects
    # is uncommon.  Without this we cannot write compressed files with our
    # own overridden info.
    zip_info._compresslevel = this_files_compress_level  # pylint: disable=protected-access
    zip_info.external_attr = desired_permissions
    zip_file.writestr(zip_info, data)
```

There is no reason for `._compresslevel` to be protected with an _. We should make it public as it is useful.

### Has this already been discussed elsewhere?

This is a minor feature, which does not need previous discussion elsewhere

### Links to previous discussion of this feature:

_No response_

### Linked PRs
* gh-113969

### PR DESCRIPTION
**Pull request by gpshead**
Make `zipfile.ZipInfo.compress_level` public.
    
A property is used to retain the behavior of the `._compresslevel`.
    
People constructing `zipfile.ZipInfo` instances to pass into existing APIs to control per-file compression levels already treat this as public, there was never a reason for it not to be.

I used the more modern name `compress_level` instead of `compresslevel` as the keyword argument on other `ZipFile` APIs is called to be consistent with `compress_type` and a general long term preference of not runningwordstogether without a separator in names.  The `zipfile` module APIs already have a variety of style inconsistencies, I don't think this makes anything worse. Someone typing the wrong one in either place will get an immediate exception.

----
📚 Documentation preview 📚: https://cpython-previews--113969.org.readthedocs.build/

* Issue: gh-113971

Fixes #113971
Closes #113971

### CODE REVIEW

**Review by erlend-aasland — APPROVED** [review]

File: Lib/test/test_zipfile/test_core.py [hunk 1]
```diff
@@ -645,6 +645,17 @@ def test_add_file_after_2107(self):
             zinfo = zipfp.getinfo(TESTFN)
             self.assertEqual(zinfo.date_time, (2107, 12, 31, 23, 59, 59))
 
+    def test_ZipInfo_compresslevel_property(self):
```
Comment (serhiy-storchaka) [inline Lib/test/test_zipfile/test_core.py]: `AbstractTestsWithSourceFile` is a wrong place for this test. `ZipInfoTests` is perhaps a correct place.

**Review by serhiy-storchaka — COMMENTED** [review]
Does it need a What's New entry?

File: Lib/test/test_zipfile/test_core.py (hunk 1 above)
Comment (gpshead) [reply Lib/test/test_zipfile/test_core.py]: good catch, thanks!

**Comment by gpshead** [comment]
> Does it need a What's New entry?

I don't think so for this one, the old way to set this attribute will continue to work forever and ZipInfo isn't even properly fully documented so a what's new entry seems like too much attention.

**Review by serhiy-storchaka — APPROVED** [review]
LGTM.

**Pull request merged**